sky/utils/kubernetes_utils.py

```diff
@@ -603,30 +611,110 @@ def __str__(self):
         return self.name
 
 
-def get_ssh_proxy_command(private_key_path: str, sshjump_name: str,
-                          namespace: str) -> str:
+def _get_proxy_command(network_mode: KubernetesNetworkingMode,
+                       private_key_path: str,
+                       ssh_jump_port: int,
+                       ssh_jump_ip: str,
+                       port_forward_proxy_cmd_path: str = '') -> str:
+    if network_mode == KubernetesNetworkingMode.NODEPORT:
+        ssh_jump_proxy_command = (f'ssh -tt -i {private_key_path} '
+                                  '-o StrictHostKeyChecking=no '
+                                  '-o UserKnownHostsFile=/dev/null '
+                                  f'-o IdentitiesOnly=yes -p {ssh_jump_port} '
+                                  f'-W %h:%p sky@{ssh_jump_ip}')
+    else:  # network_mode == KubernetesNetworkingMode.PORT_FORWARD:
+        ssh_jump_proxy_command = (
+            f'ssh -tt -i {private_key_path} '
+            f'-o ProxyCommand=\'{port_forward_proxy_cmd_path}\' '
+            '-o StrictHostKeyChecking=no '
+            '-o UserKnownHostsFile=/dev/null '
+            f'-o IdentitiesOnly=yes -p {ssh_jump_port} '
+            f'-W %h:%p sky@{ssh_jump_ip}')
+    return ssh_jump_proxy_command
```

Comment:
Can this method can be written more generally (`network_mode` and `port_forward_proxy_cmd_path` are redundant)? 

Also consider renaming to `construct_ssh_jump_command` to avoid conflict between name (`_get_proxy_command `) and arg (`port_forward_proxy_cmd_path`)
```suggestion
def construct_ssh_jump_command(private_key_path: str,
                               ssh_jump_port: int,
                               ssh_jump_ip: str,
                               proxy_cmd_path: Optional[str] = None) -> str:
    ssh_jump_proxy_command = (f'ssh -tt -i {private_key_path} '
                              '-o StrictHostKeyChecking=no '
                              '-o UserKnownHostsFile=/dev/null '
                              f'-o IdentitiesOnly=yes -p {ssh_jump_port} '
                              f'-W %h:%p sky@{ssh_jump_ip}')
    if proxy_cmd_path is not None:
        ssh_jump_proxy_command += f'-o ProxyCommand=\'{proxy_cmd_path}\' '
    return ssh_jump_proxy_command
```